packages/node_modules/@webex/plugin-meetings/src/meetings/index.js

```diff
@@ -55,48 +55,74 @@ export default class Meetings extends StatelessSparkPlugin {
    * In current nomenclature, this would create a `call`.
    * `spaceID` is used in {https://github.com/webex/spark-js-sdk/blob/master/packages/node_modules/@ciscospark/plugin-phone/src/call.js#L1007 `call.dial`}
    * @param {string} destination - {https://www.cisco.com/c/en/us/td/docs/ios-xml/ios/voice/cube/configuration/cube-book/sip-supp-uri-based-dialing.html sipURL} or spaceID, AKA {https://developer.webex.com/endpoint-rooms-roomId-get.html roomID}
-   * @returns {Meeting} A new Meeting.
+   * @returns {Promise} A new Meeting.
    */
   create(destination) {
     return this.meetingInfo
       .fetchMeetingInfo(destination)
-      .then(() => {
-        // handle response
-      })
-      .catch(() => {}) // handle errors
-      .then(() => {
-        const meeting = new Meeting(
-          {
-            sipUri: destination,
-            userId: this.spark.internal.device.userId,
-            deviceUrl: this.spark.internal.device.url,
-            roapSeq: 0,
-            meetingInfo: {}
-          },
-          {
-            parent: this.spark
-          }
-        );
+      .then((info) => info)
+      .catch((err) => err)
+      .then((info) => this.createMeeting(info, destination))
+      .catch((err) => err);
+  }
 
-        return MeetingCollection.set(meeting);
+  /**
+   * @param {String} type a string representing the type of destination, such as SIP_URI,MEETING_ID,LOCUS_ID,PERSONAL_ROOM,MEETING_LINK,ONE_ON_ONE,MEDIA_SIP_URI,CONVERSATION_URL,TEMP_SIP_URI
+   * @param {String} destination the actual mapped string for the type, i.e., we expect that if type is SIP_URI, the destination string WILL be a valid sip uri
+   * @returns {Promise} a new meeting
+   * See create() for description
+   */
+  createWithType(type, destination) {
+    return this.meetingInfo
+      .fetchMeetingInfo(destination, type)
+      .then((info) => info)
+      .catch((err) => err)
+      .then((info) => this.createMeeting(info, destination))
+      .catch((err) => err);
+  }
 
-        // Create the meeting calling the necessary service endpoints.
+  /**
+   * @param {String} info see create()
+   * @param {String} destination see create()
+   * @returns {Meeting} a new meeting instance complete with meeting info and destination
+   */
+  createMeeting(info, destination) {
+    const mtgInfo = info ? info.body : null;
+    const meeting = new Meeting(
+      // TODO: code review check: lets look at how this was structured in code review, does it make sense to set sip URI at this stage?
+      // or should it be done at the same stage as set MeetingInfo? after the meeting object has been created? because in some cases
+      // the sip uri is not available right away
+      {
+        sipUri: destination,
+        userId: this.spark.internal.device.userId,
+        deviceUrl: this.spark.internal.device.url,
+        roapSeq: 0,
+        meetingInfo: mtgInfo || {}
```

Comment:
just change line 90 to be an empty object  in the else and you can pass the full `meetingInfo` object.

Reply:
D'oh!